### PR TITLE
Update SdkServiceLoader loading

### PR DESCRIPTION
**Pull request by zoewangg**
## Description
Use the class loader that loaded the SDK to load the HTTP implementations. See #56

## Testing
`mvn clean install`

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](../docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by zoewangg** [comment]
Interceptors are loaded using `ClassLoaderHelper`. I will update `SdkServiceLoader` to use it as well. Not sure where classloading is used in sts credentials provider though.

 https://github.com/aws/aws-sdk-java-v2/blob/e00b75797b1f94789e6105913396c0c1bed3e258/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/ClassLoaderHelper.java#L117

**Comment by zoewangg** [comment]
Updated the `ClassLoader#classLoader` to load the class via parent class loaders rather than the context loader

**Comment by varunnvs92** [comment]
what is motivation for the change? I mean what is advantage of using sdk class loader vs context class loader - performance?

**Comment by zoewangg** [comment]
@varunnvs92  In some environments, loading classes from the thread context class loader will cause issues.

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/ClassLoaderHelper.java [hunk 1]
```diff
@@ -25,22 +25,20 @@ private ClassLoaderHelper() {
 
     private static Class<?> loadClassViaClasses(String fqcn, Class<?>[] classes) {
         if (classes != null) {
```
Comment (varunnvs92) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/ClassLoaderHelper.java]: null check is redundant as its now done in classLoader method

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/ClassLoaderHelper.java [hunk 2]
```diff
@@ -25,22 +25,20 @@ private ClassLoaderHelper() {
...
-            for (Class<?> c : classes) {
```
Comment (varunnvs92) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/ClassLoaderHelper.java]: The behavior seems different. In old code, we would use class loader of each class if loading class doesn't work. But in new code, we only use the loader for the first class that returns non-null getClassLoader.

Comment (zoewangg) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/ClassLoaderHelper.java]: Oops.. Good catch, will fix!

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/ClassLoaderHelper.java (hunk 1 above)
Comment (zoewangg) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/util/ClassLoaderHelper.java]: I'd still prefer leaving it here coz we might add other caller methods w/o the null check in the future.

**Pull request merged**